custom_components/tahoma/water_heater_devices/domestic_hot_water_production.py

```diff
@@ -87,36 +87,69 @@ async def async_set_temperature(self, **kwargs):
 
     async def async_set_operation_mode(self, operation_mode):
         """Set new target operation mode."""
+        if operation_mode == OverkizCommandParam.BOOST:
+            await self.executor.async_execute_command(
+                OverkizCommand.SET_BOOST_MODE, OverkizCommand.ON
+            )
+            return
+        if operation_mode != OverkizCommandParam.BOOST and self._is_boost_mode_on:
```

Comment:
The first condition will be always True. If we are in boost mode we enter into the previous `if` block.

Reply:
fixed